modules/alloy/src/test/ts/browser/behaviour/ReplacingTest.ts

```diff
@@ -1,249 +1,633 @@
+                      specB,
+                      premadeB,
+                      specA
+                    ]
+                  }
+                ]);
+
+                Assertions.assertStructure(
+                  'Checking structure span(specB, premadeB, specA)',
+                  ApproxStructure.build((s, str, arr) => {
+                    return s.element('div', {
+                      children: [
+                        s.element('span', {
+                          children: [
+                            s.element('span', { classes: [ arr.has('b') ] }),
+                            s.element('span', { classes: [ arr.has('premade-b') ] }),
+                            s.element('div', { classes: [ arr.has('a') ] })
+                          ]
+                        })
+                      ]
+                    });
+                  }),
+                  spec.comp.element
+                );
+
+                Replacing.set(spec.comp, [
+                  {
+                    dom: {
+                      tag: 'span',
+                    },
+                    components: [
+                      premadeB,
+                      specA
+                    ]
+                  }
+                ]);
+
+                Assertions.assertStructure(
+                  'Checking structure span(premadeB, specA)',
+                  ApproxStructure.build((s, str, arr) => {
+                    return s.element('div', {
+                      children: [
+                        s.element('span', {
+                          children: [
+                            s.element('span', { classes: [ arr.has('premade-b') ] }),
+                            s.element('div', { classes: [ arr.has('a') ] })
+                          ]
+                        })
+                      ]
+                    });
+                  }),
+                  spec.comp.element
+                );
+              }
+            );
+          })
+        ),
+
+        sCheckReplaceAt(
+          '.replaceAt 0 of 0 - should do nothing',
+          spec.comp,
+          [],
+          [],
+          0,
+          Optional.some('replaceAt-0')
+        ),
 
+        sCheckReplaceAt(
+          '.replaceAt 0 of 1',
+          spec.comp,
+          [ 'replaceAt-0' ],
+          [ 'original' ],
+          0,
+          Optional.some('replaceAt-0')
+        ),
+
+        sCheckReplaceAt(
+          '.replaceAt 0 of 3',
+          spec.comp,
+          [ 'replaceAt-0', 'original2', 'original3' ],
+          [ 'original1', 'original2', 'original3' ],
+          0,
+          Optional.some('replaceAt-0')
+        ),
+
+        sCheckReplaceAt(
+          '.replaceAt 2 of 3',
+          spec.comp,
+          [ 'original1', 'original2', 'replaceAt-2' ],
+          [ 'original1', 'original2', 'original3' ],
+          2,
+          Optional.some('replaceAt-2')
+        ),
+
+        sCheckReplaceAt(
+          '.replaceAt 2 with noting',
```

Comment:
I think this is a typo?
```suggestion
          '.replaceAt 2 with nothing',
```

Reply:
Good catch thanks